### PR TITLE
ITKDev - patron page

### PR DESCRIPTION
**Pull request by sinejespersen**
#### Link to issue

n/a

#### Description

Patron page module

**Remember to change react branch**

#### Screenshot of the result

If your change affects the user interface you should include a screenshot of the result with the pull request.

#### Checklist

- [ ] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [ ] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [ ] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

If you have any further comments or questions for the reviewer them please add them here.

### CODE REVIEW

**Review by cableman — APPROVED** [review]

**Review by cableman — APPROVED** [review]

**Review by cableman — APPROVED** [review]

**Review by cableman — APPROVED** [review]

File: web/modules/custom/dpl_patron_page/src/Plugin/Block/PatronPageBlock.php [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  /**
+   * Checks whether the library has enabled text messages.
+   *
+   * @return string
+   *   True if enabled, false if disabled.
+   */
+  private function textNotificationsEnabled(): string {
+    $patron_page_settings = $this->configFactory->get('patron_page.settings');
+    return empty($patron_page_settings->get('text_notifications_enabled')) ? 'false' : 'true';
+  }
```
Comment (spaceo) [inline web/modules/custom/dpl_patron_page/src/Plugin/Block/PatronPageBlock.php:75]: We have been using "1"/"0" inother cases. I think we should stick to one way of repesenting boolean in config

File: web/modules/custom/dpl_patron_page/src/Plugin/Block/PatronPageBlock.php [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    $dateConfig = $general_config->get('pause_reservation_start_date_config');
+    if (is_null($dateConfig)) {
+      $dateConfig = "";
+    }
+
+    $data = [
+      // Configuration.
+      'text-notifications-enabled-config' => $this->textNotificationsEnabled(),
+      'blacklisted-pickup-branches-config' => DplReactAppsController::buildBranchesListProp($this->branchSettings->getExcludedReservationBranches()),
+      'branches-config' => DplReactAppsController::buildBranchesJsonProp($this->branchRepository->getBranches()),
```
Comment (spaceo) [inline web/modules/custom/dpl_patron_page/src/Plugin/Block/PatronPageBlock.php:98]: This should really be a service. But I understand that you stick for this for now.
Consider writing a comment about it.

**Review by spaceo — CHANGES_REQUESTED** [review]
Hi Looks good. I have one comment about the Patron access check though.

File: web/modules/custom/dpl_patron_page/src/Plugin/Block/PatronPageBlock.php (hunk 1 above)
Comment (sinejespersen) [reply web/modules/custom/dpl_patron_page/src/Plugin/Block/PatronPageBlock.php:75]:
```suggestion
    return empty($patron_page_settings->get('text_notifications_enabled')) ? '0' : '1';
```

wie so?

File: web/modules/custom/dpl_patron_page/src/Controller/DplPatronPageController.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   */
+  public function profile(): array {
+    /** @var \Drupal\dpl_patron_page\Plugin\Block\PatronPageBlock $plugin_block */
+    $plugin_block = $this->blockManager->createInstance('dpl_patron_page_block', []);
+
+    $access_result = $plugin_block->access($this->currentUser());
+    if (is_object($access_result) && $access_result->isForbidden() || is_bool($access_result) && !$access_result) {
+      throw new AccessDeniedHttpException();
+    }
```
[the remark this answers is not in the document]
Comment (sinejespersen) [reply web/modules/custom/dpl_patron_page/src/Controller/DplPatronPageController.php:61]: I've added a todo.

File: web/modules/custom/dpl_patron_page/src/Plugin/Block/PatronPageBlock.php (hunk 1 above)
Comment (spaceo) [reply web/modules/custom/dpl_patron_page/src/Plugin/Block/PatronPageBlock.php:75]: Si

**Pull request merged**